CRM/Civicase/Test/Fabricator/CaseCategoryInstance.php

```diff
@@ -0,0 +1,48 @@
+<?php
+
+use CRM_Civicase_Test_Fabricator_CaseCategory as CaseCategoryFabricator;
+
+/**
+ * CaseCategoryInstance Fabricator class.
+ */
+class CRM_Civicase_Test_Fabricator_CaseCategoryInstance {
+
+  /**
+   * Fabricate a case category instance.
+   *
+   * @param array $params
+   *   Parameters.
+   *
+   * @return array
+   *   Results.
+   */
+  public static function fabricate(array $params = []) {
+    $params = self::mergeDefaultParams($params);
+    $result = civicrm_api3('CaseCategoryInstance', 'create', $params);
+
+    return array_shift($result['values']);
+  }
+
+  /**
+   * Merge to default parameters.
+   *
+   * @param array $params
+   *   Parameters.
+   *
+   * @return array
+   *   Resulting merged parameters.
+   */
+  private static function mergeDefaultParams(array $params) {
+    $defaultParams = [];
+    if (!isset($params['category_id'])) {
+      $defaultParams['category_id'] = CaseCategoryFabricator::fabricate()['value'];
+    }
+    if (!isset($params['instance_id'])) {
+      $instanceClasses = CRM_Core_OptionGroup::values('case_category_instance_type');
```

Comment:
Can we rename this variable? The function is returning option values with the value column as the key and the `label` column as the label. So we can rename to `$caseCategoryInstanceOptions` or something better.

Reply:
@tunbola Good point, and actually, the variable plays no role there. I just removed it: https://github.com/compucorp/uk.co.compucorp.civicase/pull/657/commits/bc2196e4d34a04c7e79b59c5fb8d7c32a7ec1924